hw/ip/spi_host/doc/_index.md

```diff
@@ -146,32 +151,117 @@ A preliminary investigation of DTR transfer mode suggests that proper support fo
+For such devices, all commands can in principle be treated as bidirectional, as both the host and device are always transmitting on their respective lines.
+For bidirectional commands, the SPI_HOST IP will store one byte in the RX FIFO for each byte transmitted from the TX FIFO.
+
+However, even for these standard SPI commands, software may be uninterested in some or all of the device response data.
+For example, for SPI flash devices, standard-mode write commands contain no useful data in the device response, even though the device may be actively asserting signals to `sd[1]` throughout the transaction.
+Therefore, for such commands software may choose to specify the entire command as "Tx Only", in which case data placed in the TX FIFO will be transmitted throughout the write command, but signals received from the device will be ignored and will not fill the RX FIFO.
+
+Meanwhile for other flash commands, such as standard-mode read, the device only transmits useful information during some portions of the transaction.
+In the case of a basic read (with a 3-byte address), the instruction starts with a 1-byte instruction code (0x3) followed by the three address bytes, during which time the flash device outputs may be high impedance (depending on the device).
+The device then immediately responds with the requested data in the next `sck` cycle, and continues to output data bytes until the `csb` line is deasserted.
+Though such a command could also be treated as entirely bidirectional, the device response can be safely ignored during the instruction and address phase (especially if the sd[1] line is high impedance during this time). 
+Likewise it is not necessary for software to specify any data to transmit while the device is responding.
+Therefore such a command can be thought of as consisting of two separate segments, the first segment being TX Only and the second segment being Rx only, as shown in the following figure.
+Breaking the command up this way potentially simplifies the job of writing software for this type of command.
+
+{{< wavejson >}}
+{signal: [
+  {name: "clk_core_i",     wave: "p....................|.............."},
+  {name: "sck (CPOL=0)",   wave: "0.1010101010101010101|01010101010101"},
+  {name: "csb",            wave: "10...................|.............."},
+  {name: "sd[0]",          wave: "00.0.0.0.0.0.1.1.2.2.|2.2.x.........", data: ["a[23]", "a[22]", "a[1]", "a[0]"]},
+  {name: "sd[1]",          wave: "z....................|....2.2.2.2.2.", data: ["d[7]", "d[6]", "d[5]", "d[4]", "..."]},
+  {name: "Segment number", wave: "x2...................|....2.........", data: ['1', '2', '3','4'] },
+  {name: "Segment speed",  wave: "x2...................|....2.........", data: ['Standard', 'Standard'] },
+  {name: "Segment Direction", wave: "x2...................|....2.........", data: ['TX', 'RX', 'None', 'RX'] },
```

Comment:
Direction -> direction (for consistency with other names above)? (Here and in other waveforms as well.)

Reply:
Ah! good catch. Thanks.